counsel.el

```diff
@@ -1438,22 +1438,48 @@ This function should set `ivy--old-re'."
 
 (defun counsel-git-grep-action (x)
   "Go to occurrence X in current Git repository."
-  (when (string-match "\\`\\(.*?\\):\\([0-9]+\\):\\(.*\\)\\'" x)
-    (let ((file-name (match-string-no-properties 1 x))
-          (line-number (match-string-no-properties 2 x)))
+  (let ((file-and-line-number (counsel--git-grep-file-and-line-number x)))
+    (when file-and-line-number
       (find-file (expand-file-name
-                  file-name
+                  (car file-and-line-number)
                   (ivy-state-directory ivy-last)))
-      (goto-char (point-min))
-      (forward-line (1- (string-to-number line-number)))
-      (when (re-search-forward (ivy--regex ivy-text t) (line-end-position) t)
-        (when swiper-goto-start-of-match
-          (goto-char (match-beginning 0))))
-      (swiper--ensure-visible)
-      (run-hooks 'counsel-grep-post-action-hook)
-      (unless (eq ivy-exit 'done)
-        (swiper--cleanup)
-        (swiper--add-overlays (ivy--regex ivy-text))))))
+      (counsel--git-grep-go-to-location (cdr file-and-line-number)))))
+
+(defun counsel-git-grep-action-other-window (x)
+  "Go to occurrence X in current Git repository in another window."
+  (let ((file-and-line-number (counsel--git-grep-file-and-line-number x)))
+    (when file-and-line-number
+      (find-file-other-window (expand-file-name
+                               (car file-and-line-number)
+                               (ivy-state-directory ivy-last)))
+      (counsel--git-grep-go-to-location (cdr file-and-line-number)))))
+
+(rx-define grouped-regex-matches
```

Comment:
[ As an aside, regexp names should follow the same conventions as function names, so if we were to go ahead with `rx-define` it would be nice to name it something like `counsel--git-grep-file-and-line`. ]